metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MySqlLookupService.java

```diff
@@ -173,53 +172,48 @@ public String getValue(final String name) {
      * @return returns the lookup with the given name.
      */
     @Override
-    @Transactional
     public Lookup setValues(final String name, final Set<String> values) {
         try {
-            return jdbcUtil.getTransactionTemplate().execute(new TransactionCallback<Lookup>() {
-                @Override
-                public Lookup doInTransaction(final TransactionStatus status) {
-                    Lookup lookup = null;
-                    try {
-                        lookup = findOrCreateLookupByName(name);
-                    } catch (SQLException e) {
-                        throw new CannotCreateTransactionException(
-                            "findOrCreateTagItemByName failed " + name, e);
-                    }
-                    final Set<String> inserts;
-                    Set<String> deletes = Sets.newHashSet();
-                    final Set<String> lookupValues = lookup.getValues();
-                    if (lookupValues == null || lookupValues.isEmpty()) {
-                        inserts = values;
-                    } else {
-                        inserts = Sets.difference(values, lookupValues).immutableCopy();
-                        deletes = Sets.difference(lookupValues, values).immutableCopy();
-                    }
-                    lookup.setValues(values);
-                    if (!inserts.isEmpty()) {
-                        insertLookupValues(lookup.getId(), inserts);
-                    }
-                    if (!deletes.isEmpty()) {
-                        deleteLookupValues(lookup.getId(), deletes);
-                    }
-                    return lookup;
-                }
-            });
-        } catch (CannotCreateTransactionException | DataAccessException e) {
+            Lookup lookup = null;
+            try {
+                lookup = findOrCreateLookupByName(name);
+            } catch (SQLException e) {
+                throw new CannotCreateTransactionException(
+                    "findOrCreateTagItemByName failed " + name, e);
+            }
```

Comment:
Previously, the doInTransaction clause is used and throws this when the exception happens inside. Since that explicit transaction is removed, we can throw or propagate the SQLException instead of throw a new CannotCreateTransactionException.